src/adapter/src/coord/timestamp_selection.rs

```diff
@@ -220,6 +221,7 @@ pub trait TimestampProvider {
         compute_instance: ComputeInstanceId,
         timeline_context: TimelineContext,
         real_time_recency_ts: Option<mz_repr::Timestamp>,
+        isolation_level: &IsolationLevel,
```

Comment:
This is fine. I'm also wonder if a `Option<IsolationLevel>` is better here to make it simpler for tests and other places to do the correct thing and use the session's. Maybe:

```
override_isolation_level: Option<IsolationLevel>
```

Reply:
that seems more complex to me. `determine_timestamp` already has to consult the session vars for the metric generation so it always knows the isolation_level. The test seems like similarly complex with or without it.